src/averageAges.js

```diff
@@ -32,12 +43,23 @@ function calculateMenAverageAge(people, century) {
  * her as mother.
  *
  * @param {object[]} people
- * @param {boolean} withChildren - optional
+ * @param {boolean} withChild - optional
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChild) {
+  let womenFilter = people.filter(person => person.sex === 'f');
+  const motherName = people.map(person => person.mother);
+
+  if (withChild) {
+    womenFilter = people.filter(person => person.sex === 'f'
+      && motherName.includes(person.name));
+  }
+
+  const ages = womenFilter.map(woman => woman.died - woman.born);
+  const agesArr = ages.reduce((sum, age) => sum + age, 0);
+
+  return agesArr / ages.length;
```

Comment:
Same about the ternary operator and a separate function for average ages goes here